src/urllib3/_base_connection.py

```diff
@@ -62,6 +65,7 @@ def __init__(
             socket_options: _TYPE_SOCKET_OPTIONS | None = ...,
             proxy: Url | None = None,
             proxy_config: ProxyConfig | None = None,
+            idle_timeout: int | None = None,
```

Comment:
Most timeouts in urllib3 are floats. Why would this be a int? Could we also accept a timedelta directly? We're documenting it on L55 as a timedelta so I would also expect to pass it as that potentially.